src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -119,6 +119,8 @@ public IHostBuilder ConfigureContainer<TContainerBuilder>(Action<HostBuilderCont
 
         /// <summary>
         /// Run the given actions to initialize the host. This can only be called once.
+        /// Adds host application lifetime service to the host.
```

Comment:
It adds other services as well, for example logging, configuration, etc. Why are the lifetime services special here?

If we want to list the services added here, I think it would be more appropriate to add them to a `<remarks>` section instead.

Reply:
Lifetime services are special because they prevent the application from quitting and you must explicitly ask the application to quit from client code, whereas all other services are used as needed and do not affect the main workflow.  That is, when you tell the application to run as a background service, nothing tells you that it will run indefinitely and this is not what you, as a whatever-else programmer, would expect (a whatever-else programmer would expect an explicit call to `EnterMainEventLoop` or whatever).  I tell you this as a whatever-else programmer.  Surprise factor 90%.

Reply:
The current worker template has the following code in Program.cs:

```C#
using WorkerService6;

IHost host = Host.CreateDefaultBuilder(args)
    .ConfigureServices(services =>
    {
        services.AddHostedService<Worker>();
    })
    .Build();

await host.RunAsync();
```

The equivalent place to `EnterMainEventLoop` is the last line: `await host.RunAsync();`. This is the place where your program will run indefinitely until something cancels it.

Reply:
Yes, and this is unexpected because the code in the `Worker` provides no indication of an infinite loop and the callback method in `Worker` is called only once.  This is why it looks odd at the first encounter.

Reply:
The template has a loop in it:

```C#
namespace WorkerService6
{
    public class Worker : BackgroundService
    {
        private readonly ILogger<Worker> _logger;

        public Worker(ILogger<Worker> logger)
        {
            _logger = logger;
        }

        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
        {
            while (!stoppingToken.IsCancellationRequested)
            {
                _logger.LogInformation("Worker running at: {time}", DateTimeOffset.Now);
                await Task.Delay(1000, stoppingToken);
            }
        }
    }
}
```

The idea being that this background task loops until it is done with its work.

Check out https://docs.microsoft.com/en-us/dotnet/core/extensions/workers, which describes how all the pieces fit together.

Reply:
I think it would be fine to add a general statement to the `<remarks>` section that said something like:

```
Adds basic services to the host such as application lifetime, host environment, and logging.
```

Reply:
> The template has a loop in it:

I did not use the template because I did not know it exists; the documentation for [BackgroundService](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.backgroundservice) does not mention it at all.  Do you think it should?  I just inherited from [BackgroundService](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.backgroundservice), implemented a straightforward procedure and was stupefied by the effect.

Reply:
I think it would be reasonable for the documentation for BackgroundService to point to https://docs.microsoft.com/dotnet/core/extensions/workers.

@gewarren @IEvangelist - do we have any examples of the xml docs linking to the conceptual docs? Here's a case where I think it would be helpful.

Reply:
I'm not actually sure, but I do agree that would be helpful. Ideally, it would exclude the locale as you've done there. Checkout this [grep.app search results](https://grep.app/search?q=docs.microsoft.com/en-us/dotnet&filter[repo][0]=dotnet/runtime).

Reply:
You can add a `<related>` tag like [this](https://github.com/dotnet/runtime/blob/b2ab353c0ed8eb9d394570a7cde8c4ed03c771e5/src/libraries/System.Private.CoreLib/src/System/Numerics/Vector2.cs#L691), which adds a link under the **See also** heading.